File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.py
class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        summ = 0

        max_sum = -math.inf

        for i in range(k):
            summ += nums[i]

        max_sum = (max_sum if max_sum > summ else summ)

        for i in range(k ,len(nums)):
            summ += nums[i]
            summ -= nums[i - k]

            max_sum = (max_sum if max_sum > summ else summ)

        return max_sum / k




        

File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.java
class Solution {
    public double findMaxAverage(int[] nums, int k) {
        int maxSum = 0, winSum = 0;

        for(int i = 0; i < k; i++){
            maxSum += nums[i];
        }

        winSum = maxSum;

        for(int i = k; i < nums.length; i++){
            winSum += nums[i] - nums[i - k];
            maxSum = Math.max(maxSum, winSum);
        }

        double res = maxSum / (double) k;
        return res;
    }
}